R/translate-sql-conditional.R

```diff
@@ -31,8 +31,14 @@ sql_if <- function(cond, if_true, if_false = quo(NULL), missing = quo(NULL)) {
   sql(paste0(out, " END"))
 }
 
-sql_case_when <- function(...) {
+sql_case_when <- function(..., .default = NULL, .ptype = NULL, .size = NULL) {
   # TODO: switch to dplyr::case_when_prepare when available
+  if (!is_null(.ptype)) {
```

Comment:
Pass along `error_call` so the error message reports the error is coming from `case_when()`?  `mssql_case_when()` also needs similar treatment, although that's really getting into the weeds.

Reply:
`mssql_case_when()` uses `sql_case_when()`, so this works directly.